docs/standards/python_coding_standards.md

```diff
@@ -0,0 +1,304 @@
+        ...
+    ```
+    _In this example, naming clearly indicates purpose: `DataProcessor` is a class, `DataLoadError` is an exception, `MAX_CONNECTIONS` is a constant, and load_data is a function. Variables like `source` and `result_list` use descriptive snake_case names._
+
+## Import Styles and Structure
+Organise imports in a consistent, readable manner. Our import guidelines (aligning with PEP 8 and GDS) are:
+
+- Import Order: Group imports into three sections, each separated by a blank line:
+
+    1. Standard library imports (e.g. os, sys, json).
+    2. Third-party library imports (e.g. fastapi, sqlalchemy, langchain).
+    3. Local application imports (your own modules/packages).
+    This makes it easy to see which dependencies are standard vs external vs internal.
+
+- One Import Per Line: Import one module per line. Avoid combining unrelated imports on the same line. 
+    
+    Example:
+    ``` python
+    import os  
+    import sys  
+    from datetime import datetime
+    ```
+    (Each import on its own line is clearer and makes diffs cleaner.)
+
+- Absolute Imports: Prefer absolute imports (full path to modules) for clarity. For example, use `from myapp.utils import helpers` instead of relative imports like `from . import helpers`, unless the relative import greatly improves clarity in a complex package. (Standard library code always uses absolute imports.) In our projects, absolute imports are usually straightforward and preferred.
+
+- Avoid Wildcard Imports: Do not use `from module import *`. Wildcard imports make it unclear which names are present and can lead to collisions. Instead, explicitly import what you need or import the module and use `module.name`. The only exception to this rule is if a module’s `__init__.py` uses wildcard import to expose its API (even then, that's handled within the package, not in application code).
+
+- Import Aliases: Only alias imports if necessary to avoid name conflicts or for clarity. For example, if you import a module with a long name, you might do `import transformations_pipeline as pipeline` for convenience, but generally keep the original names for transparency.
+
+- Ordering Within Sections: Within each import group, you can order alphabetically by module name to keep them sorted (our linter will do this automatically, minimising merge conflicts). For example, in third-party imports, `import numpy` should come before `import requests` (alphabetical).
+
+    Import Ordering Example at top of a module:
+    ``` python
+    """Module for item routes and operations (routes/items.py)."""
+
+    # Standard library imports
+    import json
+    import os
+    from pathlib import Path
+
+    # Third-party imports
+    import requests
+    from fastapi import APIRouter, HTTPException
+
+    # Local application imports
+    from app import config
+    from app.models import Item
+    from app.utils import calculate_score
```

Comment:
This is pleasing to read for me as this is how I order module imports in Node.js.  However, that is not a consistent approach taken across Defra and it's not something we mandate in the equivalent standards.

So I'm conflicted between whether this is a good idea because as a new language we've got more of an opportunity to be more opinionated from the beginning on lower level code choices, or if this is just going too low and will just lead to friction and debate for a hill I've got no passion to die on.

Reply:
I agree, this does feel a bit like stepping over the bounds for a standard, but then the Python ecosystem is a bit different to that for Node (where everything is equal) and .NET (where everything by Microsoft is equal).

There's Python modules that are built in C and are a rock-solid underpinning of things like data science, which are maybe a bit different to some package that a hobbyist knocked together at the weekend.

But I'm not a Python expert, so perhaps I'm not aware of some specific value in mandating this?